public/jsx/views/login.jsx

```diff
@@ -12,50 +13,41 @@ var gettext = require('utils').gettext;
 module.exports = React.createClass({
 
   displayName: 'LoginView',
-
   mixins: [Navigation],
+  UserActions: UserActions,
+  UserStore: UserStore,
 
-  getInitialState: function() {
-    return {
-      is_logged_in: false,
-    };
+  contextTypes: {
+    router: React.PropTypes.func,
   },
 
   componentDidMount: function() {
     var { router } = this.context;
-    $.ajax({
-      data: {
-        access_token: router.getCurrentQuery().access_token,
-      },
-      method: 'post',
-      url: '/api/auth/sign-in/',
-      context: this,
-    }).then(function(data) {
-      if (this.isMounted()) {
-        this.setState({'is_logged_in': true}); // eslint-disable-line
-
-        UserActions.setCurrentUser({email: data.buyer_email});
-
-        console.log('setting CSRF token for subsequent requests:', data.csrf_token);
-        $.ajaxSetup({
-          headers: {
-            "X-CSRFToken": data.csrf_token,
-          },
-        });
-
-        this.transitionTo('card-listing');
-      }
-    }).fail(function() {
-      // TODO: some error state.
-      console.log('login failed');
-    });
+    this.UserStore.addSetUserListener(this.onSetUser);
+    this.UserActions.signIn(router.getCurrentQuery().access_token);
   },
 
-  contextTypes: {
-    router: React.PropTypes.func,
+  componentWillUnmount: function() {
+    this.UserStore.removeSetUserListener(this.onSetUser);
+  },
+
+  onSetUser: function() {
+    if (!this.isMounted()) {
+      console.log('ignoring events while unmounted');
+      return;
+    }
+    var user = this.UserStore.getCurrentUser();
```

Comment:
This will raise an error if the user hasn't been set; not sure the conditional logic in the following lines matters at all. I think you want to try/catch instead?